types/src/circom.rs

```diff
@@ -0,0 +1,210 @@
+    zkid::{
+        ZkIdPublicKey, ZkIdSignature, MAX_EPK_BYTES, MAX_EXPIRY_HORIZON_SECS, MAX_ISS_BYTES,
+        MAX_JWT_HEADER_BYTES,
+    },
+};
+use aptos_crypto::{poseidon_bn254, CryptoMaterialError};
+use ark_bn254::{Fq, Fq2, G1Affine, G2Affine};
+use ark_ff::PrimeField;
+use ark_groth16::{PreparedVerifyingKey, VerifyingKey};
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+
+pub static DEV_VERIFYING_KEY: Lazy<PreparedVerifyingKey<ark_bn254::Bn254>> = Lazy::new(dev_pvk);
+
+fn dev_pvk() -> PreparedVerifyingKey<ark_bn254::Bn254> {
+    // Convert the projective points to affine.
+    let alpha_g1 = G1Projective::new(
+        "16672231080302629756836614130913173861541009360974119524782950408048375831661",
+        "1076145001163048025135533382088266750240489485046298539187659509488738517245",
+    )
+    .to_affine()
+    .unwrap();
+
+    let beta_g2 = G2Projective::new(
+        [
+            "1125365732643211423779651913319958385653115422366520671538751860820509133538",
+            "10055196097002324305342942912758079446356594743098794928675544207400347950287",
+        ],
+        [
+            "10879716754714953827605171295191459580695363989155343984818520267224463075503",
+            "440220374146936557739765173414663598678359360031905981547938788314460390904",
+        ],
+    )
+    .to_affine()
+    .unwrap();
+
+    let gamma_g2 = G2Projective::new(
+        [
+            "10857046999023057135944570762232829481370756359578518086990519993285655852781",
+            "11559732032986387107991004021392285783925812861821192530917403151452391805634",
+        ],
+        [
+            "8495653923123431417604973247489272438418190587263600148770280649306958101930",
+            "4082367875863433681332203403145435568316851327593401208105741076214120093531",
+        ],
+    )
+    .to_affine()
+    .unwrap();
+
+    let delta_g2 = gamma_g2;
+
+    let mut gamma_abc_g1 = Vec::new();
+    for points in [
+        G1Projective::new(
+            "10630119204695129176884860852234232187032863639334371023708138007302523646865",
+            "8100947059469766601395165113187306282631271312167186605231839390439402060594",
+        )
+        .to_affine()
+        .unwrap(),
+        G1Projective::new(
+            "18669717593291583006164561820680929698908561353625908867516300854867219058689",
+            "8091804270019087529935049146021494025057159496668931947922664231857415567945",
+        )
+        .to_affine()
+        .unwrap(),
+    ] {
+        gamma_abc_g1.push(points);
+    }
+
+    let vk = VerifyingKey {
+        alpha_g1,
+        beta_g2,
+        gamma_g2,
+        delta_g2,
+        gamma_abc_g1,
+    };
+
+    PreparedVerifyingKey::from(vk)
+}
+
+fn parse_field_element(s: &str) -> Result<Fq, CryptoMaterialError> {
+    s.parse::<Fq>()
+        .map_err(|_e| CryptoMaterialError::DeserializationError)
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash, Serialize)]
+pub struct G1Projective {
```

Comment:
This format is not efficient. We need to store G1 and G2 bytes as compressed bytes.

Let's discuss asynchronously.

Reply:
Updated to take 32 and 64 byte representations of G1 and G2 projective respectively.